packages/gantt/frontend/sirius-components-gantt/src/helper/helper.tsx

```diff
@@ -180,3 +189,87 @@ export const getDisplayedColumns = (gqlColumns: GQLColumn[]): Column[] => {
   return gqlColumns.filter((col) => col.displayed === true).map((col) => TaskListColumnEnum[col.id]);
 };
+
+const getDayOfTheYear = (date: Date) => {
+  const start = new Date(date.getFullYear(), 0, 0);
+  const diff = date.getTime() - start.getTime() + (start.getTimezoneOffset() - date.getTimezoneOffset()) * 60 * 1000;
+  const oneDay = 1000 * 60 * 60 * 24;
+  const day = Math.floor(diff / oneDay);
+  return day;
+};
+
+export const roundDate = (
+  date: Date,
+  dateExtremity: DateExtremity,
+  action: BarMoveAction,
+  dateMoveStep: String
+): Date => {
+  const regex = /^(\d+)([DHm])$/;
+  const matches = dateMoveStep.match(regex);
+
+  if (matches && matches[1]) {
+    let value = parseInt(matches[1], 10);
+    const dimension = matches[2];
+    const newdate = new Date(date);
+    if (dimension == 'D') {
+      let dayOfTheYear: number = getDayOfTheYear(newdate);
+      if (newdate.getMinutes() != 0 || newdate.getHours() != 0 || (dayOfTheYear - 1) % value > 0) {
+        newdate.setMinutes(0);
+        newdate.setHours(0);
+        if (dateExtremity == 'start' || action == 'move') {
+          // In case of move we need to round start and end date with the same direction (floor) so that the duration keeps unchanged
+          dayOfTheYear = Math.floor((dayOfTheYear - 1) / value) * value + 1; // OK
+        } else if (dateExtremity == 'end') {
+          dayOfTheYear = Math.ceil(dayOfTheYear / value) * value + 1; // OK
+        }
+        newdate.setMonth(0);
+        newdate.setDate(1);
+        newdate.setDate(dayOfTheYear);
+      }
+    } else if (dimension == 'H') {
+      if (newdate.getMinutes() != 0 || newdate.getHours() % value > 0) {
+        newdate.setMinutes(0);
+        let hours = newdate.getHours();
+        if (dateExtremity == 'start' || action == 'move') {
+          // In case of move we need to round start and end date with the same direction (floor) so that the duration keeps unchanged
+          hours = Math.floor(newdate.getHours() / value) * value;
+        } else if (dateExtremity == 'end') {
+          hours = Math.ceil(newdate.getHours() / value) * value;
+        }
+        newdate.setHours(hours);
+      }
+    } else if (dimension == 'm') {
+      if (newdate.getMinutes() % value > 0) {
+        let minutes = newdate.getMinutes();
+        if (dateExtremity == 'start' || action == 'move') {
+          // In case of move we need to round start and end date with the same direction (floor) so that the duration keeps unchanged
+          minutes = Math.floor(newdate.getMinutes() / value) * value;
+        } else if (dateExtremity == 'end') {
+          minutes = Math.ceil(newdate.getMinutes() / value) * value;
+        }
+        newdate.setMinutes(minutes);
+      }
+    }
+    newdate.setSeconds(0);
+    return newdate;
+  } else {
+    return new Date(date);
+  }
+};
+
+export const checkIsHoliday = (date: Date, _, __, dateExtremity: DateExtremity) => {
```

Comment:
What is supposed to be the definition of a holiday? We have end users pretty much in every continent and people may not agree on your definition of holiday.

Reply:
Currently, holiday is hard coded for Saturday and Sunday.
I have made some changes in gantt-task-react so that it would work fine if holiday is implemented differently by the client (sirius-web). But, the feature to delegate the holiday definition to the specifier is not done in sirius-web.
I will add a comment.